### PR TITLE
docs/flow: explain how to reference a component's exports

### PR DESCRIPTION
**Pull request by rfratto**
Give a detailed explanation for how to reference the exports of a component.

### CODE REVIEW

File: docs/flow/concepts/configuration_language.md [hunk 1]
```diff
@@ -61,6 +61,13 @@ supports more complex expressions, such as:
 
 Expressions may be used for any attribute inside a component definition.
 
+### Referencing component exports
+
+The most common expression is to reference the exports of a component like
+`local.file.password_file.content`. A reference to a component's exports is
+formed by merging the component's name (e.g., `local.file`), label (e.g.,
```
Comment (mattdurham) [inline docs/flow/concepts/configuration_language.md:68]: should we add `label if exists`?

Comment (rfratto) [reply docs/flow/concepts/configuration_language.md:68]: Good point. It feels a little weird to squeeze into the sentence, but I'll add a second sentence for components that don't use labels

File: docs/flow/concepts/configuration_language.md [hunk 2]
```diff
@@ -61,6 +61,17 @@ supports more complex expressions, such as:
...
+`password_file`), and export name (e.g., `content`), delimited by period.
+
+For components that don't use labels, like
+`prometheus.integration.node_exporter`, only combine the component name with
+export name: `prometheus.integration.node_exporter.targets`.
```
Comment (rfratto) [inline docs/flow/concepts/configuration_language.md:73]: We don't actually have a merged component which doesn't use a label right now, so we might need to remove this by v0.28 if #1872 doesn't get merged.

**Review by mattdurham — APPROVED** [review]
LGTM

**Pull request merged**